### DISCUSSION TITLE
Running so-saltstack-update : should I use 2.4/main or 2.4.50 branch?

### DISCUSSION THREAD

**Discussion by DoughnutPeach** [Security-Onion-Solutions/securityonion#12552] [2.4] [answered]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

upgrading

### Installation Type

Distributed

### Location

cloud

### Hardware Specs

Exceeds minimum requirements

### CPU

32

### RAM

16

### Storage for /

500GB

### Storage for /nsm

1TB

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I recently ran `so-saltstack-update` by accident, without specifying the branch. This caused a couple of errors and I managed to recover using my backup salt configs (ref: https://github.com/Security-Onion-Solutions/securityonion/discussions/12495).

Since then, salt has been giving me a couple of problems and I believe the root cause of it is that some files are still runing v2.3.

I would like to re-run `so-saltstack-update` to reconfigure everything to v2.4. But should I use 2.4/main or 2.4.50 branch? Thanks alot for your help in advance!

 
## Context
Faced a couple of problems and I suspect there are still some v2.3 scripts in my environment.

i.e. scripts not updating, as mentioned here (https://github.com/Security-Onion-Solutions/securityonion/discussions/12495#discussioncomment-8687759).

I'm also seeing the following errors when opening SOC > Configuration in `/opt/so/log/soc/sensoroni-server.log`. I noticed that the `grains.host` error seems to be something [v2.3](https://github.com/Security-Onion-Solutions/securityonion/blob/2.3.290/salt/idh/defaults/defaults.yaml) uses but not [v2.4.50](https://github.com/Security-Onion-Solutions/securityonion/blob/2.4.50/salt/idh/defaults.yaml). 
```
{
  "fields": {
    "error": "yaml: invalid map key: map[string]interface {}{\"grains.host\":interface {}(nil)}",
    "path": "/opt/so/saltstack/default/salt/idh/defaults.bak/defaults.yaml"
  },
  "level": "error",
  "timestamp": "2024-03-11T09:50:05.813614132Z",
  "message": "Unable to parse YAML file"
}
{
  "fields": {
    "error": "yaml: invalid map key: map[string]interface {}{\"grains.host\":interface {}(nil)}",
    "requestId": "9d486aec-e09f-4179-a258-63af87af9c9f",
    "requestor": {
      "id": "8f04f15f-342a-4e10-86a7-bd15538599e9",
      "createTime": "2024-03-11T09:50:05.745478297Z",
      "updateTime": "0001-01-01T00:00:00Z",
      "email": "xxxxxx",
      "firstName": "",
      "lastName": "",
      "totpStatus": "",
      "oidcStatus": "",
      "webauthnStatus": "",
      "note": "",
      "roles": null,
      "status": "",
      "searchUsername": "",
      "password": "",
      "passwordChanged": false
    }
  },
  "level": "warn",
  "timestamp": "2024-03-11T09:50:05.815627581Z",
  "message": "Request did not complete successfully"
}
```

Thus I highly suspect that there are some v2.3 scripts that were not removed in my environment.. I hope I'm not going crazy.

Thanks in advance for any help! Appreciate it!

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by dougburks** [comment, marked as answer]
This question is answered at https://github.com/Security-Onion-Solutions/securityonion/discussions/12495#discussioncomment-8761794. 

WARNING! For others that may come across this discussion, please note that `so-saltstack-update` is an undocumented command and is not used under normal circumstances. It is primarily a developer tool and should only be used if you are explicitly directed to do so.
